src/components/VideoCanvas/VideoCanvas.tsx

```diff
@@ -398,9 +401,18 @@ export const VideoCanvas = memo(forwardRef<VideoRef, VideoProps>((props, ref) =>
 
   useEffect(() => {
     let isLoaded = false;
+    let timeout = undefined;
 
-    const checkVideoLoaded = () => {
+    const checkVideoLoaded: (counter: number) => NodeJS.Timeout | undefined = (counter) => {
       if (isLoaded) return;
+      if (counter > 300) {
```

Comment:
300 is some what jarring here - would it be possible to create a constant variable so it describes what it is?